pkg/plugin/keyvault.go

```diff
@@ -93,94 +91,99 @@ func NewKeyVaultClient(
-		vaultURL = getProxiedVaultURL(vaultURL, proxyAddress, proxyPort)
+	keyIDHash, err := getKeyIDHash(vaultURL, keyName, keyVersion)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get key id hash, error: %w", err)
 	}
 
-	mlog.Always("using kms key for encrypt/decrypt", "vaultURL", *vaultURL, "keyName", keyName, "keyVersion", keyVersion)
+	mlog.Always("using kms key for encrypt/decrypt", "vaultURL", vaultURL, "keyName", keyName, "keyVersion", keyVersion)
 
 	client := &KeyVaultClient{
-		baseClient:       kvClient,
-		config:           config,
-		vaultName:        vaultName,
-		keyName:          keyName,
-		keyVersion:       keyVersion,
-		vaultURL:         *vaultURL,
-		azureEnvironment: env,
-		keyIDHash:        keyIDHash,
+		baseClient: kvClient,
+		config:     config,
+		vaultName:  vaultName,
+		keyName:    keyName,
+		keyVersion: keyVersion,
+		keyIDHash:  keyIDHash,
 	}
 	return client, nil
 }
 
+type policyFunc func(req *policy.Request) (*http.Response, error)
+
+func (p policyFunc) Do(req *policy.Request) (*http.Response, error) {
+	return p(req)
+}
+
+var _ policy.Policy = (*policyFunc)(nil)
+
 // Encrypt encrypts the given plain text using the keyvault key.
 func (kvc *KeyVaultClient) Encrypt(
 	ctx context.Context,
 	plain []byte,
-	encryptionAlgorithm kv.JSONWebKeyEncryptionAlgorithm,
+	encryptionAlgorithm azkeys.EncryptionAlgorithm,
 ) (*service.EncryptResponse, error) {
 	value := base64.RawURLEncoding.EncodeToString(plain)
 
-	params := kv.KeyOperationsParameters{
-		Algorithm: encryptionAlgorithm,
-		Value:     &value,
+	params := azkeys.KeyOperationParameters{
+		Algorithm: &encryptionAlgorithm,
+		Value:     []byte(value),
 	}
-	result, err := kvc.baseClient.Encrypt(ctx, kvc.vaultURL, kvc.keyName, kvc.keyVersion, params)
+	result, err := kvc.baseClient.Encrypt(ctx, kvc.keyName, kvc.keyVersion, params, nil)
 	if err != nil {
 		return nil, fmt.Errorf("failed to encrypt, error: %+v", err)
 	}
 
-	if kvc.keyIDHash != fmt.Sprintf("%x", sha256.Sum256([]byte(*result.Kid))) {
+	if kvc.keyIDHash != fmt.Sprintf("%x", sha256.Sum256([]byte(*result.KID))) {
 		return nil, fmt.Errorf(
 			"key id initialized does not match with the key id from encryption result, expected: %s, got: %s",
 			kvc.keyIDHash,
-			*result.Kid,
+			*result.KID,
 		)
 	}
 
 	annotations := map[string][]byte{
-		dateAnnotationKey:           []byte(result.Header.Get(dateAnnotationValue)),
-		requestIDAnnotationKey:      []byte(result.Header.Get(requestIDAnnotationValue)),
-		keyvaultRegionAnnotationKey: []byte(result.Header.Get(keyvaultRegionAnnotationValue)),
-		versionAnnotationKey:        []byte(encryptionResponseVersion),
-		algorithmAnnotationKey:      []byte(encryptionAlgorithm),
+		// dateAnnotationKey:           []byte(result.Header.Get(dateAnnotationValue)),
+		// requestIDAnnotationKey:      []byte(result.Header.Get(requestIDAnnotationValue)),
+		// keyvaultRegionAnnotationKey: []byte(result.Header.Get(keyvaultRegionAnnotationValue)),
```

Comment:
This is a breaking change compared to the upstream repo and it might've implications on OCP support